src/main/java/world/data/jdbc/internal/query/SparqlEngine.java

```diff
@@ -104,8 +104,10 @@ public ParameterMetaData getParameterMetaData(String query) throws SQLException
     @Override
     public ResultSet execute(DataWorldStatement statement, String query, Map<String, Node> parameters, Integer timeoutSeconds)
             throws SQLException {
+        Integer maxRowsToReturn = statement.getMaxRows() != 0 ? statement.getMaxRows() : null;
+
         // Execute the query
-        Response response = queryApi.executeQuery(query, parameters, timeoutSeconds);
+        Response response = queryApi.executeQuery(query, parameters, maxRowsToReturn, timeoutSeconds);
```

Comment:
nit: @Nullable on `QueryApi.executeQuery()` maxRowsToReturn param

Reply:
Could you just pass down `statement.getMaxRows()` and then do a check for 0 in HttpQueryApi?

Reply:
Imho the HttpQueryApi shouldn't need to know anything about "0 means unlimited", it's an artifact of the sql statement api.